d2bs/kolbot/libs/bots/TeamDiablo.js

```diff
@@ -0,0 +1,462 @@
+/**
+ *    @filename    Teamdiablo.js
+ *    @author      Jaenster
+ *    @desc        Make a true party of a baalrun.
+ */
+
+/*
+ */
+function TeamDiablo() {
+    var diablo = {
+        run: function () {
+            // Init the team, if not already done
+            Team.init();
+            Team.msg.requestData('diablo');
+
+            // Get to throne
+            if (Config.TeamDiablo.Teleport && !Team.communication.diablo.portalReady) {
+                diablo.common.teleport();
+            } else {
+                Team.util.goBo();
+                Town.doChores();
+                Town.goToTown(4)
+            }
+
+            if (!Config.TeamDiablo.Teleport && Team.util.doingBaal() && Config.TeamDiablo.TeleportToThrone) {
+                diablo.util.print('Teleporting to throne');
+                Pather.useWaypoint(Areas.WorldstoneLvl2);
+
+                if (!Pather.moveToExit([Areas.WorldstoneLvl3, Areas.ThroneOfDestruction], true)) {
+                    throw new Error("Failed to move to Throne of Destruction.");
+                }
+
+                Pather.moveTo(Team.data.baalPortalSpot[0],Team.data.baalPortalSpot[1]);
+                Pather.makePortal(true);
+
+                Team.communication.baal.AtWorldstoneLvl3 = true;
+                Team.communication.baal.portalReady = true;
+                Team.msg.sendData('baal'); // send update so everyone know the portal is ready
+            }
+
+            // Run all seals
+            var who;
+            for (who in diablo.data.seals) {
+                if (Config.TeamDiablo[who.toLowerCase()] && !Team.communication.diablo[who + 'Done'] && !Team.communication.diablo.diabloDone) {
+                    this.common.doSeal(who);
+                }
+            }
+
+            // In the you define teams to do the seals, but a teammember is missing, make sure we did all the seals
+            if (Config.TeamDiablo.Viz || Config.TeamDiablo.Seis || Config.TeamDiablo.Inf) {
+
+                // Either one is defined we should do. So make sure we did all. Or help others
+                while (!Team.communication.diablo.vizDone || !Team.communication.diablo.seisDone || !Team.communication.diablo.vizDone) {
+
+                    for (who in diablo.data.seals) {
+                        if (!Team.communication.diablo[who + 'Done'] && !Team.communication.diablo.diabloDone) {
+                            this.common.doSeal(who);
+                        }
+                    }
+                }
+            }
+
+            if (Config.TeamDiablo.Diablo && !Team.communication.diablo.diabloDone) {
+                this.common.killDia();
+            }
+
+            if (!Team.communication.diablo.diabloDone) {
+                Town.goToTown(4);
+
+                while (!baal.data.communication.diabloDone) {
+                    // ToDo: Should make this more fail proof
+                    delay(1000);
+                }
+            }
+        },
+        common: {
```

Comment:
Don't think the common object is necessary, the member functions are not used in other files.